hashednode.go

```diff
@@ -44,14 +44,8 @@ func (HashedNode) Get([]byte, NodeResolverFn) ([]byte, error) {
 	return nil, errors.New("can not read from a hash node")
 }
 
-func (HashedNode) Commit() *Point {
-	// TODO: we should reconsider what to do with the VerkleNode interface and how
-	//       HashedNode fits into the picture, since Commit(), Commitment() and Hash()
-	//	     now panics. Despite these calls must not happen at runtime, it is still
-	//	     quite risky. The reason we end up in this place is because PBSS came quite
-	//	     recently compared with the VerkleNode interface design. We should probably
-	//	     reconsider splitting the interface or find some safer workaround.
-	panic("can not commit a hash node")
+func (n HashedNode) Commit() (*Point, error) {
+	return n.Commitment(), nil
```

Comment:
this is a no-op change, and it removes some comments that are useful in case we want to resurrect this repo later.